src/proxy-ui-api/src/test/java/org/niis/xroad/restapi/service/KeyServiceTest.java

```diff
@@ -72,6 +80,23 @@ public void setup() throws Exception {
         KeyInfo keyInfo = TokenTestUtils.createTestKeyInfo(GOOD_KEY_ID);
         tokenInfo.getKeyInfo().add(keyInfo);
         when(signerProxyFacade.getTokens()).thenReturn(Collections.singletonList(tokenInfo));
+
+        doAnswer(invocation -> {
+            Object[] arguments = invocation.getArguments();
+            String newKeyName = (String) arguments[1];
+            ReflectionTestUtils.setField(keyInfo, "friendlyName", newKeyName);
+            return null;
+        }).when(signerProxyFacade).setKeyFriendlyName(any(), any());
+
+        doAnswer(invocation -> {
+            Object[] arguments = invocation.getArguments();
+            String newKeyName = (String) arguments[1];
+            if("new-friendly-name-update-fails".equals(newKeyName)) {
+                throw new CodedException(SIGNER_X + "." + X_KEY_NOT_FOUND);
+            }
+            return null;
+        }).when(signerProxyFacade).setKeyFriendlyName(anyString(), eq("new-friendly-name-update-fails"));
```

Comment:
Related to the earlier comment above. This string could be checked in the other `setKeyFriendlyName` mock above